src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/IncludeRewriter.cs

```diff
@@ -27,19 +28,17 @@ public override Expression VisitSqlRoot(SqlRootExpression expression, object con
                 return expression;
             }
 
-            bool containsInclude = false;
+            bool containsInclude = expression.TableExpressions.AsEnumerable().Where(e => e.SearchParameterQueryGenerator is IncludeQueryGenerator).Any();
 
-            List<TableExpression> reorderedExpressions = expression.TableExpressions.OrderByDescending(t =>
+            if (!containsInclude)
             {
-                switch (t.SearchParameterQueryGenerator)
-                {
-                    case IncludeQueryGenerator _:
-                        containsInclude = true;
-                        return 0;
-                    default:
-                        return 10;
-                }
-            }).ToList();
+                return new SqlRootExpression(expression.TableExpressions, expression.DenormalizedExpressions);
+            }
+
+            // Order expressions so that simple search parameters appear first and include parameters appear after
+            // In addition, sort that _include:iterate and _revinclude:iterate parameters are ordered so that the
+            // included results they reference appears before
+            List<TableExpression> reorderedExpressions = expression.TableExpressions.OrderByDescending(t => t, new TableExpressionComparer()).ToList();
```

Comment:
I realize it was like this before, but it's a tad confusing that the comparer reverses the return value and we are ordering by descending.

Reply:
Also please reuse a static instance of TableExpressionComparer

Reply:
I agree it's counter intuitive, I was trying to be consistent with prev code.
I can definitely remove the OrderByDescending and reverse the comparer accordingly.

Reply:
Both fixed.